### PR TITLE
fix plugin bug

### PR DESCRIPTION
**Pull request by Stevent-fei**
### Describe what this PR does / why we need it
When running the plugin script, the values in the plugins do not take effect.
I have found the cause of the problem: there is a problem with the repeated judgment function in plugins, which causes the plugin to not execute.
so i fix it.

### CODE REVIEW

**Comment by kakaZhou719** [comment]
@Stevent-fei ，could you pls describe this bug more specifically which is  helpful for others to read it.

File: pkg/plugin/plugins.go [hunk 1]
```diff
@@ -74,30 +74,26 @@ func (c *PluginsProcessor) Load() error {
...
-				return err
-			}
-			Register(pt, p)
-		}
-		if yaml.Matcher(f.Name()) {
-			plugins, err := utils.DecodeCRDFromFile(filepath.Join(path, f.Name()), common.Plugin)
-			if err != nil {
-				return fmt.Errorf("failed to load plugin: %v", err)
+
+	if files != nil {
```
Comment (Starnop) [inline pkg/plugin/plugins.go:78]: if the files ==nil, 

the logic like `for _, f := range files` will not be executed actually

So no need to change that, just use `for _, f := range files` is better

File: pkg/plugin/plugins.go [hunk 2]
```diff
@@ -74,30 +74,26 @@ func (c *PluginsProcessor) Load() error {
...
+		c.Plugins = nil
```
Comment (Starnop) [inline pkg/plugin/plugins.go:79]: why reset c.Plugins here?

**Comment by Starnop** [comment]
image

It is better to use `git commit-s --amend` or `git rebase with squash`

File: pkg/plugin/plugins.go [hunk 3]
```diff
@@ -87,23 +86,36 @@ func (c *PluginsProcessor) Load() error {
...
+			c.Plugins = append(c.Plugins, plugins.([]v1.Plugin)...)
+			i := deduplication(c.Plugins)
+			c.Plugins = i
 		}
 	}
 
 	return nil
 }
 
+func deduplication(plugins []v1.Plugin) []v1.Plugin {
```
Comment (Starnop) [inline pkg/plugin/plugins.go:101]: so what has changed

Comment (Stevent-fei) [reply pkg/plugin/plugins.go:101]: The previous function did not do all the deduplication

**Comment by 27149chen** [comment]
what is the status of this pr?

**Comment by Stevent-fei** [comment]
> 

@27149chen The plugin bug has been fixed.

**Comment by 27149chen** [comment]
> > 
> 
> @27149chen The plugin bug has been fixed.

@Stevent-fei in which version? I didn't find any newer releases?

**Comment by Stevent-fei** [comment]
> > > 
> > 
> > 
> > @27149chen The plugin bug has been fixed.
> 
> @Stevent-fei in which version? I didn't find any newer releases.

@27149chen this change has not been merged, you can use this branch to build new binaries

**Comment by 27149chen** [comment]
> > > > 
> > > 
> > > 
> > > @27149chen The plugin bug has been fixed.
> > 
> > 
> > @Stevent-fei in which version? I didn't find any newer releases.
> 
> @27149chen this change has not been merged, you can use this branch to build new binaries

sorry, I can not use a private branch which is not reviewed or approved by anyone

**Review by kakaZhou719 — APPROVED** [review]
LGTM

**Pull request merged**